From fff1e3b978b1a1a75cc9c4f85acd76dd9ec89613 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Tue, 29 Apr 2025 09:08:56 +0300
Subject: [PATCH] [SW-226884] gaudi3: scripts: add 325/328 eeprom update

In order to convert hl325<->hl328 product number in eeprom

Change-Id: Ief8c8ed5660f0e067fba14db47a7ba3a2da8953a
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/scripts/make/gaudi3/Makefile b/scripts/make/gaudi3/Makefile
index c04fab0..cd8c2e6 100644
--- a/scripts/make/gaudi3/Makefile
+++ b/scripts/make/gaudi3/Makefile
@@ -92,6 +92,26 @@
 gaudi3-eeprom-create-default: gaudi3-eeprom-set-defaults gaudi3-eeprom-create
 	@echo "default eeprom created"
 
+.PHONY: gaudi3-eeprom-set-hl325
+gaudi3-eeprom-set-hl325:
+	@echo "System Product HL=145" > $(IMAGES_DIR)/eeprom.txt
+
+.PHONY: gaudi3-eeprom-create-hl325
+gaudi3-eeprom-create-hl325: gaudi3-eeprom-set-hl325
+	@$(HOST_MAKE_ENV) ${BR2_EXTERNAL}/board/habanalabs/common/create-eeprom-fit.sh
+	@mv $(ITBS_DIR)/${HL_DEV}-eeprom.itb $(ITBS_DIR)/${HL_DEV}-eeprom-hl325.itb
+	@echo "HL325 eeprom created"
+
+.PHONY: gaudi3-eeprom-set-hl328
+gaudi3-eeprom-set-hl328:
+	@echo "System Product HL=148" > $(IMAGES_DIR)/eeprom.txt
+
+.PHONY: gaudi3-eeprom-create-hl328
+gaudi3-eeprom-create-hl328: gaudi3-eeprom-set-hl328
+	@$(HOST_MAKE_ENV) ${BR2_EXTERNAL}/board/habanalabs/common/create-eeprom-fit.sh
+	@mv $(ITBS_DIR)/${HL_DEV}-eeprom.itb $(ITBS_DIR)/${HL_DEV}-eeprom-hl328.itb
+	@echo "HL328 eeprom created"
+
 EEPROM_DATA_FILE = $(IMAGES_DIR)/eeprom-recovery.txt
 .PHONY: gaudi3-eeprom-set-recovery
 gaudi3-eeprom-set-recovery:
@@ -109,6 +129,10 @@
 	$(HOST_MAKE_ENV) ${BR2_EXTERNAL}/board/habanalabs/common/create-eeprom-fit.sh 1 $(RECOVERY_PROP_NAME) recovery
 	@echo "recovery eeprom created"
 
+.NOTPARALLEL: gaudi3-eeprom-create-all
+.PHONY: gaudi3-eeprom-create-all
+gaudi3-eeprom-create-all: gaudi3-eeprom-create-hl325 gaudi3-eeprom-create-hl328 gaudi3-eeprom-create-default gaudi3-eeprom-create-recovery
+
 EFUSE_DATA_FILE = $(IMAGES_DIR)/efuse_spare_bit_write.txt
 .PHONY: gaudi3-efuse-write-set-default
 gaudi3-efuse-write-set-default:
@@ -123,6 +147,8 @@
 	@echo "default efuse write created"
 
 GAUDI3_INTERNAL_FITS := ${DEV_PREFIX}-eeprom.itb			\
+	${DEV_PREFIX}-eeprom-hl325.itb			\
+	${DEV_PREFIX}-eeprom-hl328.itb			\
 	${DEV_PREFIX}-eeprom-recovery.itb		\
 	${DEV_PREFIX}-security_enable.itb		\
 	${DEV_PREFIX}-secureboot_config.itb		\
@@ -140,8 +166,7 @@
 
 
 .PHONY: gaudi3-internal-fit-create
-gaudi3-internal-fit-create: gaudi3-eeprom-create-default	\
-				gaudi3-eeprom-create-recovery			\
+gaudi3-internal-fit-create: gaudi3-eeprom-create-all	\
 				gaudi3-security-enable-create			\
 				gaudi3-secureboot-config-create			\
 				gaudi3-secureboot-config-dbg-create		\
